src/IndexedDB.ts

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2020 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/* Simple wrapper around IndexedDB.
+ */
+
+let idb = null;
+
+async function idbInit(): Promise<void> {
+    idb = await new Promise((resolve, reject) => {
+        const request = window.indexedDB.open("element", 1);
```

Comment:
I'd argue for the codename (ie. 'vector') here rather than leaking branding to more places

Reply:
Ack.  Maybe `matrix-react-sdk` because that's the layer that we're actually in here.